.github/workflows/delivery.yml

```diff
@@ -18,24 +25,60 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v4
 
+      - name: Check whether this event is the HEAD of main
+        continue-on-error: true
+        id: is-head-main
+        run: git rev-parse HEAD | grep -x ${{ github.sha }}
+        shell: bash
+
       - name: Docker meta
         id: meta
         uses: docker/metadata-action@v5
         with:
           images: ghcr.io/${{ github.repository }}
           tags: |
             type=semver,pattern={{major}}.{{minor}}.{{patch}}
-            type=raw,value=edge
+            type=edge,enable=${{ steps.is-head-main.outcome == 'success' }}
+            type=ref,event=branch,enable=${{ github.event_name == 'workflow_dispatch' }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
 
       - name: Login to GitHub Container Registry
         uses: docker/login-action@v3
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}      
-      
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build container and export to local Docker
+        uses: docker/build-push-action@v5
+        with:
+          context: .
+          load: true
+          tags: local/${{ github.repository }}:scan
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - name: Scan Image
+        uses: anchore/scan-action@v3
+        id: scan
+        with:
+          image: local/${{ github.repository }}:scan
+          only-fixed: true
+          fail-build: true
+          severity-cutoff: critical
+          output-format: sarif
+
+      - name: Upload Anchore scan SARIF report
+        uses: github/codeql-action/upload-sarif@v3
+        if: ${{ !cancelled() }}
+        with:
+          sarif_file: ${{ steps.scan.outputs.sarif }}
+
       - name: Build container and push to GitHub Container Registry
         uses: docker/build-push-action@v5
+        if: ${{ github.event_name != 'pull_request' }}
```

Comment:
Don't push image on scheduled trigger?

Reply:
Also excluded schedule trigger from being able to push the built image.